### PR TITLE
Revise versioning guidelines in release handbook

### PR DESCRIPTION
**Pull request by StefanoFioravanzo**
@rimolive based on our discussion let's iterate on this PR to revise the versioning guidelines

### CODE REVIEW

**Review by rimolive — APPROVED** [review]

File: releases/handbook.md [hunk 1]
```diff
@@ -223,10 +223,10 @@ From that phase and forward updates to the manifests repo must only be fixing co
 
 **Actions for other WGs:**
 
-- Provide a _git revision_ which the release team members or Manifests leads will use to update the files in the manifests repo
-- Provide an initial list of issues that the WG would like to close until the final release
+- Provide a _git revision_ which the release team members or Manifests leads will use to update the files in the manifests repo. The _git revision_ should be a proper GitHub release with a descriptive changelog. The release should either be a new major/minor/patch version or a pre-release release candidate (RC).
```
Comment (juliusvonkohout) [inline releases/handbook.md:226]: I would use the full word "repository".

File: releases/handbook.md [hunk 2]
```diff
@@ -223,10 +223,10 @@ From that phase and forward updates to the manifests repo must only be fixing co
...
+- Provide an initial list of issues that the WG would like to close until the final release and the expected version number of the final component's version.
```
Comment (juliusvonkohout) [inline releases/handbook.md:227]: Provide an initial list of issues that the working group (WG) aims to resolve before the final release, as well as the expected version number of the final component.

File: releases/handbook.md [hunk 3]
```diff
@@ -223,10 +223,10 @@ From that phase and forward updates to the manifests repo must only be fixing co
...
 - Work on closing the provided open issues
-- Ensure that future_git revisions_they provide only include bug fixes, and not new features, from the previously provided _git revision_
+- Ensure that future _git revisions_ provide only include bug fixes, and not new features, from the previously provided _git revision_. If the working group provides incremental releases (either sequential RCs or minor/patch version bumps), each should be a GitHub release with an incremental changelog. After releasing a series of RCs, the final version should contain the full changelog from all RCs combined.
```
Comment (juliusvonkohout) [inline releases/handbook.md:229]: Ensure that future _git revisions_ only include bug fixes, ...

**Review by tarilabs — APPROVED** [review]
/lgtm

thank you Release team

File: releases/handbook.md (hunk 1 above)
Comment (StefanoFioravanzo) [reply releases/handbook.md:226]: Good catch!

**Review by terrytangyuan — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**